testing/framework/test-framework.rst

```diff
@@ -8,38 +8,47 @@ SCons Testing Framework
 Introduction
 ============
 
-SCons uses extensive automated tests to ensure quality. The primary goal
-is that users be able to upgrade from version to version without
-any surprise changes in behavior.
+SCons uses extensive automated tests to ensure quality, so users can
+safely upgrade from version to version without any surprise changes in behavior.
+
+Changes to SCons code, whether fix or new feature, need testing support before
+being merged. Sometimes this might mean writing tests for an area that didn't
+have adequate, or any, testing before. Having tests isn't an
+*iron-clad policy*, but exceptions need to be okayed by the core team -
+it's good to discuss ahead of time. When in doubt, test it!
+
+This guide should provide enough information to get started. Use it together
+with existing tests to "see how it's done".  Note if you're coding in an IDE,
+you may need to add the path ``testing/framework`` to the known paths,
+or it won't recognize some symbols.
 
-In general, no change goes into SCons unless it has one or more new
-or modified tests that demonstrably exercise the bug being fixed or
-the feature being added.  There are exceptions to this guideline, but
-they should be just that, *exceptions*.  When in doubt, make sure
-it's tested.
 
 Test organization
 =================
 
-There are three types of SCons tests:
-
-*End-to-End Tests*
-   End-to-end tests of SCons are Python scripts (``*.py``) underneath the
-   ``test/`` subdirectory.  They use the test infrastructure modules in
-   the ``testing/framework`` subdirectory. They set up small complete
-   projects and call SCons to execute them, checking that the behavior is
-   as expected.
+There are three categories of SCons tests:
 
 *Unit Tests*
    Unit tests for individual SCons modules live underneath the
-   ``SCons/`` subdirectory and are the same base name as the module
-   to be tested, with ``Tests`` appended  to the basename. For example,
-   the unit tests for the ``Builder.py`` module are in the
-   ``BuilderTests.py`` script.  Unit tests tend to be based on assertions.
+   ``SCons/`` subdirectory (that is alongside the code they are testing)
+   and use the Python ``unittest`` module to perform assertion-based
```

Comment:
Should include that files have to be named *Tests.py to be found.

Reply:
I thought that's what that section was saying?

Reply:
I don't see it in this section.. it's much later in the doc..

Reply:
I'm not seeing the reason for the fuss here. This is an "about" section, not a "how to" section. I can add these but it seems like extra noise.

Reply:
You removed it from the original, which I think is useful information to have up front.